common/network-types/src/frame.rs

```diff
@@ -89,6 +107,7 @@ impl Frame {
     /// Segments this frame into a list of [segments](Segment) each of maximum sizes `mtu`.
     #[inline]
     pub fn segment(&self, mtu: u16) -> Vec<Segment> {
+        assert!(self.frame_id > 0, "frame id 0 is not allowed");
```

Comment:
Why though?

Reply:
Disallowing frame id 0 allows for better organized code when it comes to the use of atomics.